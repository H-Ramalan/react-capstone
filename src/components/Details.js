import { useEffect } from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import { FiArrowLeft } from 'react-icons/fi';
import { useSelector, useDispatch } from 'react-redux';
import { fetchCountry } from '../Redux/Countries/CountriesSlice';
import './styles/Details.css';

const Details = () => {
  const country = useSelector((state) => state.countries?.country);
  const navigate = useNavigate();
  const params = useParams();
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCountry(params.country));
  }, [dispatch]);
  return (
    <div className="details-box">
      <div className="back-btn">
        <button className="back" type="button" onClick={() => navigate('/')}>
          <FiArrowLeft />
          Back
        </button>
      </div>

      <div className="arm-name">
        <img src={country?.coatOfArms} alt="coat" className="coat-of-arm" />
        <h1 className="name">{country?.commonName}</h1>
      </div>
      <div className="details">
        <ul>
          <li>
            Continent:
            {country?.continents}
          </li>
          <li>
            Capital:
            {country?.capital}
          </li>

          <li>
            Population:
            {country?.population}
          </li>
          <li>
            Land Area:
            {country?.area}
          </li>
          <li>
            Timezone:
            {country?.timeZone}
          </li>
          <li>
            Landlocked:
            {`${country?.landlocked}`}
          </li>
          <li>
            Independent:
            {`${country?.independent}`}
          </li>
        </ul>
      </div>
    </div>
  );
};

export default Details;
