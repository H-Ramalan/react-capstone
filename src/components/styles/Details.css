* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.coat-of-arm {
  width: 10%;
}

.arm-name {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 10px 0;
}

.name {
  margin: 20px 0;
  color: black;
  font-size: 30px;
}

.back {
  padding: 10px 25px;
  border: none;
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 10%;
  margin: 0 10px;
  border-radius: 5px;
}

ul {
  display: flex;
  flex-direction: column;
  align-content: space-between;
  align-items: center;
  justify-content: center;
  list-style: none;
  font-size: 25px;
}

.details-box {
  background-color: rgb(228, 130, 130);
  height: 88vh;
}

.details {
  background: brown;
  width: 70%;
  justify-content: center;
  justify-items: center;
  margin: 0 auto;
  border-radius: 15px;
}

li {
  margin: 5px 0;
}
