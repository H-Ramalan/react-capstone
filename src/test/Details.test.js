import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Details from '../components/Details';

const mockStore = configureMockStore([thunk]);

const initialState = {
  country: {
    data: [
      {
        coatOfArms: 'test-coatOfArms-url',
        CommonName: 'Test Country',
        continent: 'Test Continent',
        capital: 'Test Capital',
        population: 'Test Population',
        area: 'Test Area',
        timezone: 'Test Timezone',
        landlocked: true,
        independent: true,
      },
    ],
  },
};

describe('Details Component', () => {
  it('renders correctly', () => {
    const store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
