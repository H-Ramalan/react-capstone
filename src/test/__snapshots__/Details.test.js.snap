// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="details-box"
  >
    <div
      class="back-btn"
    >
      <button
        class="back"
        type="button"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="19"
            x2="5"
            y1="12"
            y2="12"
          />
          <polyline
            points="12 19 5 12 12 5"
          />
        </svg>
        Back
      </button>
    </div>
    <div
      class="arm-name"
    >
      <img
        alt="coat"
        class="coat-of-arm"
      />
      <h1
        class="name"
      />
    </div>
    <div
      class="details"
    >
      <ul>
        <li>
          Continent:
        </li>
        <li>
          Capital:
        </li>
        <li>
          Population:
        </li>
        <li>
          Land Area:
        </li>
        <li>
          Timezone:
        </li>
        <li>
          Landlocked:undefined
        </li>
        <li>
          Independent:undefined
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
